ivy/functional/ivy/elementwise.py

```diff
@@ -2763,7 +2777,74 @@ def log1p(
     Returns
     -------
     ret
-        a new array containing the evaluated result for each element in x.
+        an array containing the evaluated Natural logarithm of 1 + x for each element in
+        ``x``. The returned array must have a floating-point data type determined by
+        :ref:`type-promotion`.
+
+    This function conforms to the `Array API Standard
+    <https://data-apis.org/array-api/latest/>`_. This docstring is an extension of the
+    `docstring <https://data-apis.org/array-api/latest/API_specification/generated/signatures.elementwise_functions.add.html>`_ # noqa
+    in the standard.
+
+    Both the description and the type hints above assumes an array input for simplicity,
+    but this function is *nestable*, and therefore also accepts :code:`ivy.Container`
+    instances in place of any of the arguments.
+
+    Examples
+    --------
+    With :code:`ivy.Array` input:
+
+    >>> x = ivy.array([1 , 2 ,3 ])
+    >>> y = ivy.log1p(x)
+    >>> print(y)
+    ivy.array([0.693, 1.1  , 1.39 ])
+
+    >>> x = ivy.array([0 , 1 ])
+    >>> y = ivy.zeros(2)
+    >>> ivy.log1p(x , out = y)
+    >>> print(y)
+    ivy.array([0.   , 0.693])
+
+    >>> x = ivy.array([[1.1, 2.2, 3.3], \
+                   [4.4, 5.5, 6.6]])
+    >>> ivy.log1p(x , out = x)
+    >>> print(x)
+    ivy.array([[0.742, 1.16 , 1.46 ],
+               [1.69 , 1.87 , 2.03 ]])
+
+    >>> x = ivy.array([1e-99])
+    >>> y = ivy.log(1+ x)
+    >>> print(y)
+    ivy.array([0.])
+    >>> y = x.log1p( )
+    >>> print(y)
+    ivy.array([1.e-99])
+
+    With :code:`ivy.NativeArray` input:
+
+    >>> x = ivy.native_array([10., 20.])
+    >>> y = ivy.log1p(x)
+    >>> print(y)
+    ivy.array([2.4 , 3.04])
+
+
+    With :code:`ivy.Container` input:
+
+    >>> x = ivy.Container(a=ivy.array([0., 1., 2.]), b=ivy.array([3., 4., 5.1]))
+    >>> y = ivy.log1p(x)
+    >>> print(y)
+    {
+        a: ivy.array([0., 0.693, 1.1]),
+        b: ivy.array([1.39, 1.61, 1.81])
+    }
+
+    >>> x = ivy.Container(a=ivy.array([ 1., 2.]), b=ivy.array([0., 1.]))
+    >>> y = ivy.Container.static_log1p(x)
```

Comment:
`static_log1p` examples should go here https://github.com/unifyai/ivy/blob/master/ivy/container/elementwise.py 
Also, you need to add docstrings to the `log1p` and `static_log1p` functions in [container/elementwise](https://github.com/unifyai/ivy/blob/master/ivy/container/elementwise.py) and [array/elementwise](https://github.com/unifyai/ivy/blob/master/ivy/array/elementwise.py) . Also, after you have added the changes, make sure to check of the `test-docstrings / run-docstring-tests` are passing and if not, make sure that it is not because of `log1p`.

Reply:
hello , i completed the all requested changes.
but i have wired issue when i  test-doc on this example or any other other example has using out arg

x = ivy.array([0 , 1 ])
>>> y = ivy.zeros(2)
>>> x.log1p( out = y)
>>> print(x)
it return this error: 

Failed example:
    x.log1p( out = y)
Expected nothing
Got:
    ivy.array([0.   , 0.693])

also this error when use container the output of the test is same as doc-string but some how doc-test fails :

Failed example:
    print(y)
Expected:
    {
        a: ivy.array([0., 0.693, 1.1]),
        b: ivy.array([1.39, 1.61, 1.81])
    }
Got:
    {
        a: ivy.array([0., 0.693, 1.1]),
        b: ivy.array([1.39, 1.61, 1.81])
    }


so what should i do  ? i made commit with the requested changes.

Reply:
I believe the issue is not arising now.

Reply:
Looks good otherwise. Let me know and then we can merge this.

Reply:
@RickSanchezStoic  i just checked on it, and i believe we merge it